agent-inject/agent/container_env.go

```diff
@@ -64,6 +65,48 @@ func (a *Agent) ContainerEnvVars(init bool) ([]corev1.EnvVar, error) {
 			Name:  "VAULT_CONFIG",
 			Value: b64Config,
 		})
+	} else {
+		// set up environment variables to access Vault since "vault" section may not be present in the config
+		if a.Vault.Address != "" {
+			envs = append(envs, corev1.EnvVar{
+				Name:  "VAULT_ADDR",
+				Value: a.Vault.Address,
+			})
+		}
+		if a.Vault.CACert != "" {
+			envs = append(envs, corev1.EnvVar{
+				Name:  "VAULT_CACERT",
+				Value: a.Vault.CACert,
+			})
+		}
+		if a.Vault.CAKey != "" {
+			envs = append(envs, corev1.EnvVar{
+				Name:  "VAULT_CAPATH",
+				Value: a.Vault.CAKey,
+			})
+		}
+		if a.Vault.ClientCert != "" {
+			envs = append(envs, corev1.EnvVar{
+				Name:  "VAULT_CLIENT_CERT",
+				Value: a.Vault.ClientCert,
+			})
+		}
+		if a.Vault.ClientKey != "" {
+			envs = append(envs, corev1.EnvVar{
+				Name:  "VAULT_CLIENT_KEY",
+				Value: a.Vault.ClientKey,
+			})
+		}
+		envs = append(envs, corev1.EnvVar{
+			Name:  "VAULT_SKIP_VERIFY",
+			Value: strconv.FormatBool(a.Vault.TLSSkipVerify),
+		})
+		if a.Vault.TLSServerName != "" {
+			envs = append(envs, corev1.EnvVar{
+				Name:  "VAULT_TLS_SERVER_NAME",
+				Value: a.Vault.TLSServerName,
+			})
+		}
```

Comment:
Should we also include `VAULT_NAMESPACE`?

Reply:
Good question. Namespace isn't part of the `vault` config block, only `auto_auth`. And if we set `VAULT_NAMESPACE`, it could end up overriding a `namespace` the user sets in their `auto_auth` config: https://github.com/hashicorp/vault/blob/9334ae5f338c90f9a6d65e749794593e50acd48d/command/agent.go#L367-L369. However, if they've explicitly set the namespace annotation, it does seem like a reasonable thing to add. Overall it's a difficult one to call, but it maybe feels safer to exclude it for now, to maintain the current behaviour?

Reply:
Exactly. I only targeted the things that are in the `vault` config block.